roadmap.md

```diff
@@ -0,0 +1,238 @@
+    <td>These features combined enable a way to materialize intermediate results. The analysis results can be easily reused, which reduces unnecessary ad-hoc queries, improves the performance of BI and other applications (by pulling results directly) and reduces system load (by avoiding duplicated computation), thereby improving the overall data pipeline efficiency and reducing costs. It will make TiFlash an online service.</td>
+    <td>November 2022</td>
+  </tr>
+  <tr>
+    <td>Time to live (TTL)</td>
+    <td>Support automatically deleting expired table data based on custom rules. </td>
+    <td>This feature enables automatic data cleanup in limited data archiving scenarios.</td>
+    <td>December 2022</td>
+  </tr>
+  <tr>
+    <td>Multi-value Index</td>
+    <td>Support array index.</td>
+    <td>Array is one of the commonly used data types in JSON scenarios. For inclusive queries in arrays, multi-value indexes can efficiently improve the query speed. </td>
+    <td>December 2022</td>
+  </tr>
+  <tr>
+    <td>TiFlash kernel optimization</td>
+    <td><ul><li>FastScan provides weak consistency but faster table scan capability.</li><li>Further optimize the join order, shuffle, and exchange algorithms to improve computing efficiency and boost performance for complex queries.</li><li>Add a fine-grained data sharding mechanism to optimize the <code>COUNT(DISTINCT)</code> function and high cardinality aggregation.</li></ul></td>
+    <td>Improve the basic computing capability of TiFlash, and optimize the performance and reliability of the underlying algorithms of the columnar storage and MPP engine.</td>
+    <td>December 2022</td>
+  </tr>
+  <tr>
+    <td>TiDB proxy</td>
+    <td>Implement automatic load balancing so that upgrading a cluster or modifying configurations does not affect the application. After scaling out or scaling in the cluster, the application can automatically rebalance the connection without reconnecting.</td>
+    <td>In scenarios such as upgrades and configuration changes, TiDB proxy is more business-friendly.</td>
+    <td>December 2022</td>
+  </tr>
+  <tr>
+    <td>PB-level scalability</td>
+    <td>Support huge region size</td>
+    <td>Scenarios with fast business growth and a large amount of data</td>
+    <td>January 2023</td>
+  </tr>
+  <tr>
+    <td>DDL parallel framework</td>
+    <td>Implement a single DDL parallel execution framework for DDL acceleration</td>
+    <td>Accelerate DDL in scenarios such as creating indexes for large tables, modifying a column's default value, and modifying a column type.</td>
+    <td>January 2023</td>
+  </tr>
+  <tr>
+    <td>Non-prepared Plan Cache</td>
+    <td>Support plan cache for general SQL statements in a session to save cache resources, improve the hit rate of general execution plans, and improve SQL performance.</td>
+    <td>Non-prepared plan cache. Improve real-time and throughputs of OLTP in general scenarios, save PoC time, and increase PoC win rate.</td>
```

Comment:
@easonn7 . Please help check if we need to mention "increase PoC win rate" here.